packages/vm/src/runTx.ts

```diff
@@ -300,6 +300,11 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
   let fromAccount = await state.getAccount(caller)
   const { nonce, balance } = fromAccount
 
+  // EIP-3607: Reject transactions from senders with deployed code
+  if (!fromAccount.codeHash.equals(KECCAK256_NULL)) {
+    const msg = _errorMsg('invalid sender address, address is not EOA', this, block, tx)
```

Comment:
Maybe you can directly reference the EIP here to give people a chance to read up on this who are stumbling upon the error, so 'invalid sender address, address is not EOA (EIP-3607)'?

Otherwise this will get extremely hard to interpret.